Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -473,22 +476,20 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         }
     }
 
-    private func getFaviconForCell(cell:ThumbnailCell, site: Site, profile: Profile) {
-        setDefaultThumbnailBackgroundForCell(cell)
-        guard let url = site.url.asURL else { return }
+    private func downloadFaviconsAndUpdateForSite(site: Site) {
+        guard let siteURL = site.url.asURL else { return }
 
-        FaviconFetcher.getForURL(url, profile: profile) >>== { icons in
-            if icons.count == 0 { return }
-            guard let url = icons[0].url.asURL else { return }
+        FaviconFetcher.getForURL(siteURL, profile: profile).uponQueue(dispatch_get_main_queue()) { result in
+            guard let favicons = result.successValue where favicons.count > 0 else { return }
+            guard let url = favicons.first?.url.asURL else { return }
+            guard let indexOfSite = (self.data.asArray().indexOf { $0 == site }) else {
+                return
+            }
```

Comment:
Nit: these can be combined into a single `guard` statement with the `let`s separated by commas.